src/cloudsc_python/src/cloudsc4py/framework/components.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+from __future__ import annotations
+from abc import abstractmethod
+from functools import cached_property
+from typing import Optional, TYPE_CHECKING
+
+from sympl._core.core_components import (
```

Comment:
The rationale behind using Sympl is demonstrating how it can be used to write self-contained, self-documented and inter-operable components with a well-defined interface. There are however a few core functionalities that we are using, e.g. extracting raw data from a DataArray.